### DISCUSSION TITLE
Any colab-ready notebook?

### DISCUSSION THREAD

**Discussion by Extraltodeus** [oobabooga/text-generation-webui#1311] [General]
Is there a colab-ready version to just jump into it that exists?

**Comment by eucdee** [comment]
https://github.com/eucdee/AI/blob/main/4bit_TextGen_Gdrive.ipynb

LLM text generation notebook for Google Colab
This notebook uses https://github.com/oobabooga/text-generation-webui to run conversational models in chat mode.

▶⏩Run all the cells and a public gradio URL will appear at the bottom in around 5 minutes.🤞🐱‍👤

Updates:
*updated GPTQ to latest version for Vicuna 13b 1.1 support
*added more interresting models to choose from
-reeducator/vicuna-13b-free
-OpenAssistant/oasst-sft-4-pythia-12b-epoch-3.5
-TheBloke/vicuna-13B-1.1-GPTQ-4bit-128g

**Reply by Vinitrajputt** [reply]
its not running
```
/content/drive/MyDrive/text-generation-webui
python server.py --share --model TheBloke_vicuna-13B-1.1-GPTQ-4bit-128g --model_type LLaMa --settings settings-colab.json --cai-chat --extensions gallery --wbits 4 --groupsize 128
usage: server.py
       [-h]
       [--notebook]
       [--chat]
       [--character CHARACTER]
       [--model MODEL]
       [--lora LORA [LORA ...]]
       [--model-dir MODEL_DIR]
       [--lora-dir LORA_DIR]
       [--model-menu]
       [--no-stream]
       [--settings SETTINGS]
       [--extensions EXTENSIONS [EXTENSIONS ...]]
       [--verbose]
       [--loader LOADER]
       [--cpu]
       [--auto-devices]
       [--gpu-memory GPU_MEMORY [GPU_MEMORY ...]]
       [--cpu-memory CPU_MEMORY]
       [--disk]
       [--disk-cache-dir DISK_CACHE_DIR]
       [--load-in-8bit]
       [--bf16]
       [--no-cache]
       [--xformers]
       [--sdp-attention]
       [--trust-remote-code]
       [--load-in-4bit]
       [--compute_dtype COMPUTE_DTYPE]
       [--quant_type QUANT_TYPE]
       [--use_double_quant]
       [--threads THREADS]
       [--n_batch N_BATCH]
       [--no-mmap]
       [--mlock]
       [--cache-capacity CACHE_CAPACITY]
       [--n-gpu-layers N_GPU_LAYERS]
       [--n_ctx N_CTX]
       [--llama_cpp_seed LLAMA_CPP_SEED]
       [--wbits WBITS]
       [--model_type MODEL_TYPE]
       [--groupsize GROUPSIZE]
       [--pre_layer PRE_LAYER [PRE_LAYER ...]]
       [--checkpoint CHECKPOINT]
       [--monkey-patch]
       [--quant_attn]
       [--warmup_autotune]
       [--fused_mlp]
       [--gptq-for-llama]
       [--autogptq]
       [--triton]
       [--no_inject_fused_attention]
       [--no_inject_fused_mlp]
       [--desc_act]
       [--gpu-split GPU_SPLIT]
       [--flexgen]
       [--percent PERCENT [PERCENT ...]]
       [--compress-weight]
       [--pin-weight [PIN_WEIGHT]]
       [--deepspeed]
       [--nvme-offload-dir NVME_OFFLOAD_DIR]
       [--local_rank LOCAL_RANK]
       [--rwkv-strategy RWKV_STRATEGY]
       [--rwkv-cuda-on]
       [--listen]
       [--listen-host LISTEN_HOST]
       [--listen-port LISTEN_PORT]
       [--share]
       [--auto-launch]
       [--gradio-auth GRADIO_AUTH]
       [--gradio-auth-path GRADIO_AUTH_PATH]
       [--api]
       [--api-blocking-port API_BLOCKING_PORT]
       [--api-streaming-port API_STREAMING_PORT]
       [--public-api]
       [--multimodal-pipeline MULTIMODAL_PIPELINE]
server.py: error: unrecognized arguments: --cai-chat```

**Reply by Arrow75** [reply]
Not running even after unchecking "chat"

**Reply by Sensei050** [reply]
I got the same issue i don't know what to do

**Reply by rasehum** [reply]
same, hope someone managed to fix it

**Reply by GordonFreeeman** [reply]
Just in case you guys still need a fix for this - there are actually a couple of bugs, here's how I fixed them:

First is the "--cai-chat" flag messing things up. Simply change the line to "--chat" in the code:
1. Scroll down to "3. Launch", and click the small ▶ icon left of it to display the cell code
2. Look for the line "params.add('--cai-chat')", and change that to "params.add('--chat')"

Next, launch the notebook at least once (mandatory, you need the webui folder & file structure in your GDrive), and you will hit the next roadblock, which will be the gallery bug (displaying as "FileNotFoundError: [Errno 2] No such file or directory: 'characters'). To fix it,
1. Open your GDrive, and go into the folder "text-generation-webui"
2. Delete the file "characters" (that one should be a directory, but is stored as file in GDrive, and will block the next step)
3. Upload the correct oobabooga "characters" folder (I've attached it here as zip, in case you don't have it at hand)
4. Next, download the file 
5. Open it in Notepad/Notepad++ or any other text editor. Look for the line " "chat_default_extensions": ["
6. Delete the next line (which simply reads "gallery"). It should now be:
  "chat_default_extensions": [
    ],
7. Upload the file again, overwriting the on in Google Drive.
8. Fun fact: this removes absolutely nothing, the gallery will still be working just fine

After that, the Colab should launch just fine. But - Gradio doesn't seem to like text streaming, at least here, because it was throwing errors for me. In that case, make sure you disable streaming, and launch again.

Lastly, I already had a "logs" file in my GDrive, which caused another error. In that case, simply delete the file from your GDrive (it's also in the "text-generation-webui" folder).

That did it for me, the colab launches fine now. Hope it helps, cheers! 🍻 

[characters.zip](https://github.com/oobabooga/text-generation-webui/files/12050473/characters.zip)

**Comment by Extraltodeus** [comment]
thank you!

**Comment by dan-berger-a-g** [comment]
I'm getting this error trying to use it. It just exits out on it's own : 

/content/drive/MyDrive/text-generation-webui
python server.py --share --model TheBloke_vicuna-13B-1.1-GPTQ-4bit-128g --model_type LLaMa --settings settings-colab.json --wbits 4 --groupsize 128 --extensions gallery --chat
2023-07-18 00:12:56 WARNING:The gradio "share link" feature uses a proprietary executable to create a reverse tunnel. Use it with care.
2023-07-18 00:13:01.309838: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Could not find TensorRT
2023-07-18 00:13:06 INFO:Loading settings from settings-colab.json...
2023-07-18 00:13:06 INFO:Loading TheBloke_vicuna-13B-1.1-GPTQ-4bit-128g...
2023-07-18 00:13:06 INFO:The AutoGPTQ params are: {'model_basename': 'vicuna-13B-1.1-GPTQ-4bit-128g.compat.no-act-order', 'device': 'cuda:0', 'use_triton': False, 'inject_fused_attention': True, 'inject_fused_mlp': True, 'use_safetensors': False, 'trust_remote_code': False, 'max_memory': None, 'quantize_config': None, 'use_cuda_fp16': True}
2023-07-18 00:13:07 WARNING:CUDA extension not installed.
2023-07-18 00:13:10 WARNING:The model weights are not tied. Please use the `tie_weights` method before using the `infer_auto_device` function.
^C
